corehq/apps/hqadmin/models.py

```diff
@@ -65,11 +67,22 @@ def get_latest(cls, checkpoint_id):
             return None
 
     @classmethod
-    def get_historical_max(cls, checkpoint_id):
-        try:
-            return cls.objects.filter(checkpoint_id=checkpoint_id).order_by('-seq_int')[0]
-        except IndexError:
-            return None
+    def get_historical_max(cls, checkpoint_id, by_partition=False):
+        if by_partition:
+            # limit to last 10 days
+            checkpoints = cls.objects.filter(checkpoint_id=checkpoint_id)[:10]
+            max_offsets = defaultdict(int)
+            for checkpoint in checkpoints:
+                for partition, offset in checkpoint.items():
```

Comment:
how is `checkpoint` a dict?

Reply:
you are right, should be `checkpoint.seq` updating now